### PR TITLE
Avoid computing cardinality for run container 

### PR DESCRIPTION
**Pull request by xtonik**
### SUMMARY
Avoid computing cardinality for run container using `contains()` instead of check before and after using `getCardinality()` as it is always recomputed from the scratch, not so for bitmap or array containers. I provided benchmark for this also (hoping there isn't any hidden pitfall as working with bitmap and not with container directly), there can be clearly seen that the performance is heavily improved for run containers and is the same for other types of container.

| Benchmark | (containerType) | (inputData) | Mode | Cnt | Score | Error | Units |
| --- | --- | --- | --- | --- | --- | --- | --- |
| optimized | RUN | HALF_DUPLICATED | avgt | 5 | 181.238 | ±3.275 | ms/op |
| optimized | RUN | UNIQUE | avgt | 5 | 293.722 | ±4.827 | ms/op |
| optimized | BITMAP | HALF_DUPLICATED | avgt | 5 | 178.400 | ±1.207 | ms/op |
| optimized | BITMAP | UNIQUE | avgt | 5 | 313.261 | ±2.413 | ms/op |
| original | RUN | HALF_DUPLICATED | avgt | 5 | 350.664 | ±1.275 | ms/op |
| original | RUN | UNIQUE | avgt | 5 | 458.972 | ± 28.209 | ms/op |
| original | BITMAP | HALF_DUPLICATED | avgt | 5 | 180.033 | ±2.625 | ms/op |
| original | BITMAP | UNIQUE | avgt | 5 | 312.270 | ±3.662 | ms/op |

### Automated Checks

- [x] I have run `./gradlew test` and made sure that my PR does not break any unit test.
- [x] I have run `./gradlew checkstyleMain` or the equivalent and corrected the formatting warnings reported.

### CODE REVIEW

File: jmh/src/jmh/java/org/roaringbitmap/CheckedAddBenchmark.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    if (inputData.equals("HALF_DUPLICATED")) {
+      for (int i = 0; i < valuesCount; i++) {
+        a[i] = i >> 2;
+      }
+    }
+    if (inputData.equals("UNIQUE")) {
+      for (int i = 0; i < valuesCount; i++) {
+        a[i] = i;
+      }
+    }
```
Comment (richardstartin) [inline jmh/src/jmh/java/org/roaringbitmap/CheckedAddBenchmark.java:37]: just a style note here, it's nice to define an enum with values `UNIQUE`, `HALF_DUPLICATED` with routines that produce the data, it keeps the definitions of the scenarios, the implementations of the scenarios, and the rest of the setup code separated (imagine your eyes glazing over the the branches here if there were more than two scenarios).

**Review by richardstartin — APPROVED** [review]
Seems like an excellent optimisation to me

File: jmh/src/jmh/java/org/roaringbitmap/CheckedAddBenchmark.java (hunk 1 above)
Comment (xtonik) [reply jmh/src/jmh/java/org/roaringbitmap/CheckedAddBenchmark.java:37]: I like enums very much, but I want to create benchmark simple as possible. Ok, I will introduce enum for this purpose.

Comment (xtonik) [reply jmh/src/jmh/java/org/roaringbitmap/CheckedAddBenchmark.java]: Done in last commit.

**Comment by lemire** [comment]
I agree with @richardstartin 

Approving and merging.

**Pull request merged**